src/aspire/numeric/cupy_fft.py

```diff
@@ -1,35 +1,105 @@
+import functools
+
 import cupy as cp
+import cupyx.scipy.fft as cufft
+import numpy as np
 
 from aspire.numeric.base_fft import FFT
 
 
+# This improves the flexibility of our FFT wrappers by allowing for
+# incremental code changes and testing.
+def _preserve_host(func):
+    """
+    Method decorator that returns a numpy/cupy array result when
+    passed a numpy/cupy array input respectively.
+
+    At the time of writing this wrapper will also upcast cupy FFT
+    operations to doubles as the precision in singles can cause
+    accuracy issues.
+    """
+
+    @functools.wraps(func)  # Pass metadata (eg name and doctrings) from `func`
+    def wrapper(self, x, *args, **kwargs):
+
+        # CuPy's single precision FFT appears to be too inaccurate for
+        # many of our unit tests, so the signal is upcast and recast
+        # on return.
+        # Todo, discuss with Joakim whether we want this upcasting
```

Comment:
Can remove this comment, no?

Reply:
I removed the comment.  We can hopefully revisit this after addressing areas of the code that are not very robust in singles (single precision filters for example).

Reply:
Sounds good. Should we make an issue?